src/graphql.ts

```diff
@@ -135,7 +137,7 @@ directive @computed on FIELD_DEFINITION
 
   let nodes = {} as Record<number, Node>;
 
-  function toNode<T>(vertex: Vertex): Node & T {
+  function toNode<T>(vertex: Vertex, preset: API | EmptyObject = {}): Node & T {
```

Comment:
All of the generated data should be in `vertex.data`, so no need to pass in the preset.

Reply:
@cowboyd I'm not sure I follow, this is the interface for  vertex

```ts
export interface Vertex<TData = any> {
  type: string;
  id: number;
  data: TData;
}
```

I only need the keys of `preset`, how do I get the keys of preset from that

Reply:
There are no keys, but is it not enough to check if the key of the computed is present in the data?

Reply:
@cowboyd I can't tell from the type where that is and I am very unfamiliar with the code base

Reply:
it’s basically just a untyped record. Vertex.data is just whatever the factory generated merged with the presets so if the preset is in there it means it got passed in.

Reply:
@cowboyd can you show me?  I'm not sure what you mean

Reply:
@cowboyd you have a slight advantage over me on this one as you wrote the code.  It is non-obvious to me :)

Reply:
OK, I think I know what you mean in the last commit, but it is a lot of code in 1 file that you wrote, and if it is obvious to you then that is why.

Reply:
Heh. I have zero no expectation that it should be obvious. If I'm terse it's probably because I'm typing with my thumbs on my phone :)